### PR TITLE
[l10n] Improve czech (cs-CZ) locale

### PR DESCRIPTION
**Pull request by harastaivan**
Update grammar mistakes and improve translations so it's more natural to the czech reader.

### CODE REVIEW

**Comment by harastaivan** [comment]
The label is added by maintainers, correct?

File: packages/grid/x-data-grid/src/locales/csCZ.ts [hunk 1]
```diff
@@ -72,9 +72,9 @@ const csCZGrid: Partial<GridLocaleText> = {
   filterOperatorIs: 'je',
   filterOperatorNot: 'není',
   filterOperatorAfter: 'je po',
-  filterOperatorOnOrAfter: 'je na nebo po',
+  filterOperatorOnOrAfter: 'je po včetně',
   filterOperatorBefore: 'je před',
-  filterOperatorOnOrBefore: 'je na nebo dříve',
+  filterOperatorOnOrBefore: 'je před včetně',
```
Comment (alexfauquette) [inline packages/grid/x-data-grid/src/locales/csCZ.ts:77]: @Haaxor1689 if you want to have a look since you created this file

**Review by cherniavskii — APPROVED** [review]
Thank you!

**Comment by flaviendelangle** [comment]
I'm merging, if the creator of the file has some feedback we can modify later on :+1:

**Pull request merged**